booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,651 @@
+            .formatted(contextPath));
+        }
+        if (nodeToValidate.path("numberOfPackages").isMissingNode() && nodeToValidate.path("description").isMissingNode()) {
+          return Set.of("The '%s' object did not have a 'numberOfPackages' nor an 'description', which is required due to dangerousGoods"
+            .formatted(contextPath));
+        }
+        return Set.of();
+      }
+    ),
+    JsonAttribute.allIndividualMatchesMustBeValid(
+      "DangerousGoods implies numberOfPackages or description",
+      mav -> mav.submitAllMatching("requestedEquipments.*.commodities.*.outerPackaging"),
+      (nodeToValidate, contextPath) -> {
+        var dg = nodeToValidate.path("dangerousGoods");
+        if (!dg.isArray() || dg.isEmpty()) {
+          return Set.of();
+        }
+        if (nodeToValidate.path("numberOfPackages").isMissingNode() && nodeToValidate.path("description").isMissingNode()) {
+          return Set.of("The '%s' object did not have a 'numberOfPackages' nor an 'description', which is required due to dangerousGoods"
+            .formatted(contextPath));
+        }
+        return Set.of();
+      }
+    ),
+    JsonAttribute.allIndividualMatchesMustBeValid(
+      "The 'imoClass' values must be from dataset",
+      allDg((dg) -> dg.path("imoClass").submitPath()),
+      JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.DG_IMO_CLASSES)
+    ),
+    JsonAttribute.allIndividualMatchesMustBeValid(
+      "The 'segregationGroups' values must be from dataset",
+      allDg((dg) -> dg.path("segregationGroups").all().submitPath()),
+      JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.DG_SEGREGATION_GROUPS)
+    ),
+    JsonAttribute.allIndividualMatchesMustBeValid(
+      "The 'inhalationZone' values must be from dataset",
+      allDg((dg) -> dg.path("inhalationZone").all().submitPath()),
+      JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.INHALATION_ZONE_CODE)
+    ),
+    JsonAttribute.allOrNoneArePresent(
+      JsonPointer.compile("/declaredValue"),
+      JsonPointer.compile("/declaredValueCurrency")
+    ),
+    JsonAttribute.allIndividualMatchesMustBeValid(
+      "The charges 'currencyCode' values must be from dataset",
+      mav -> mav.submitAllMatching("charges.*.currencyCode"),
+      JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.ISO_4217_CURRENCY_CODES)
+    ),
+
+    JsonAttribute.allIndividualMatchesMustBeValid(
+      "The charges currency amount must not exceed more than 2 decimal points",
+      mav -> mav.submitAllMatching("charges.*"),
+      (nodeToValidate, contextPath) -> {
+        var currencyAmount = nodeToValidate.path("currencyAmount").asDouble();
+        if (BigDecimal.valueOf(currencyAmount).scale() > 2) {
+          return Set.of("currencyAmount %s is expected to have 2 decimal precious ".formatted(contextPath));
+        }
+        return Set.of();
+      }
```

Comment:
This could be simplified even further by using `charges.*.currentAmount` rather than `charges.*` (would remove the need for the `.path` and the issue message could just use `%s` directly without having to clarify that it is `currentAmount` at `%s`, etc.)